src/server/controllers/RedirectController.ts

```diff
@@ -56,12 +56,18 @@ export class RedirectController implements RedirectControllerInterface {
       const logRedirect = () => {
         this.analyticsLogger.logRedirectAnalytics(
           req,
-          res,
           shortUrl.toLowerCase(),
           longUrl,
         )
       }
 
+      const generatedCookie = this.analyticsLogger.generateCookie(
+        req.headers.cookie,
+      )
+      if (generatedCookie) {
+        res.cookie(...generatedCookie)
+      }
```

Comment:
The logic here is a bit roundabout - basically we are saying "speculatively generate the cookie, and if we happen to have one, set it". It really should say "check for the presence of a cookie, and if absent, generate one and set it"